### DISCUSSION TITLE
3rd party stylesheet and wire:navigate

### DISCUSSION THREAD

**Discussion by jacksleight** [livewire/livewire#7714] [1. Help]
I'm using a 3rd party library (Mapbox GL) that injects a stylesheet `link` tag at the end of the body when it first loads. However when a new page is loaded via `wire:navigate` the link tag is lost and the maps lose their styling.

I can manually include the link tag myself in the head (which results in duplicate tags once Mapbox loads, but that doesn't really matter). But I was just wondering if anyone knows of a tidier solution?

I don't think there's any way to alter how Mapbox injects the element.

**Comment by danie-ramdhani** [comment]
maybe using `data-navigate-once` or inject it in the head tag
because all assets in the body are re-evaluated (https://livewire.laravel.com/docs/navigate#scripts-in-the-body-are-re-evaluated)

**Reply by jacksleight** [reply]
Yeah trouble is I'm not the one adding the tag, the 3rd party library is, so I can't just throw a `data-navigate-once` on it (or add it to the head instead).

I guess I could listen for when the tag is added and then alter/move it.
